pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -298,6 +295,54 @@ func (pl *VolumeZone) EventsToRegister() []framework.ClusterEventWithHint {
 	}
 }
 
+// getPersistentVolumeClaimNameFromPod gets pvc names bound to a pod.
+func (pl *VolumeZone) getPersistentVolumeClaimNameFromPod(pod *v1.Pod) []string {
+	var pvcNames []string
+	for i := range pod.Spec.Volumes {
+		volume := pod.Spec.Volumes[i]
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+		pvcName := volume.PersistentVolumeClaim.ClaimName
+		pvcNames = append(pvcNames, pvcName)
+	}
+	return pvcNames
+}
+
+// isSchedulableAfterStorageClassAdded is invoked whenever a StorageClass is added.
+// It checks whether the addition of StorageClass has made a previously unschedulable pod schedulable.
+// Only a new StorageClass with WaitForFirstConsumer will cause a pod to become schedulable.
+func (pl *VolumeZone) isSchedulableAfterStorageClassAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
```

Comment:
As we talked in DM, can you refactor this QHint so that it doesn't fetch any other kind of resources? Then, if you think there's nothing we can do for StorageClass, please just update the comment in EventsToRegister. 🙏

Reply:
Current implementation depends on other resources. I will modify this PR

Reply:
I've changed the logic not using other resources.
However, it is too simple, just checking sc's volumebinding mode.
Is it fine?

Reply:
Looks like when storage class is not found, volumezone rejects Pods -
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/volumezone/volume_zone.go#L144-L147

Hence, whether a storage class is `VolumeBindingWaitForFirstConsumer` or not relevant? I'm now suspecting that we can always return Queue to storage class creations.

Reply:
In my understanding, storage class's change might make a pod schedulable when pvc is bound to sc and sc's mode is `VolumeBindingWaitForFirstConsumer`.
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/volumezone/volume_zone.go#L138-L154

Therefore, I think at least sc's mode is relevant.
It is written here as well.
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/volumezone/volume_zone.go#L279-L281